testing/trino-tests/src/test/java/io/trino/security/TestAccessControl.java

```diff
@@ -1472,6 +1476,165 @@ public void testAccessControlWithRolesAndRowFilter()
         assertAccessAllowed(session, "SELECT nationkey FROM nation");
     }
 
+
+    @Test
+    public void testSchemasAuthorization()
+    {
+        reset();
+
+        String schema1 = "schema_" + randomNameSuffix();
+        String schema2 = "schema_2_" + randomNameSuffix();
+        String schemaOwnerName1 = "schema_owner_1_" + randomNameSuffix();
+        String schemaOwnerName2 = "schema_owner_2_" + randomNameSuffix();
+
+        Session schemaOwner1 = TestingSession.testSessionBuilder()
+                .setIdentity(Identity.ofUser(schemaOwnerName1))
+                .setCatalog(getSession().getCatalog())
+                .setSchema(getSession().getSchema())
+                .build();
+
+        Session schemaOwner2 = TestingSession.testSessionBuilder()
+                .setIdentity(Identity.ofUser(schemaOwnerName2))
+                .setCatalog(getSession().getCatalog())
+                .setSchema(getSession().getSchema())
+                .build();
+
+        getQueryRunner().execute(
+                schemaOwner1,
+                "CREATE SCHEMA memory.%s".formatted(schema1));
+        assertQuery(
+                "SELECT * FROM system.metadata.schemas_authorization",
+                "VALUES('memory', '%s', 'USER', '%s')".formatted(schema1, schemaOwnerName1));
+
+        getQueryRunner().execute(schemaOwner1, "ALTER SCHEMA memory.%s SET AUTHORIZATION %s".formatted(schema1, schemaOwnerName2));
+        assertQuery(
+                "SELECT * FROM system.metadata.schemas_authorization",
+                "VALUES('memory', '%s', 'USER', '%s')".formatted(schema1, schemaOwnerName2));
+
+        getQueryRunner().execute(
+                schemaOwner1,
+                "CREATE SCHEMA memory_test.%s".formatted(schema2));
+        assertQuery(
+                "SELECT * FROM system.metadata.schemas_authorization",
+                "VALUES('memory', '%s', 'USER', '%s'), ('memory_test', '%s', 'USER', '%s')".formatted(schema1, schemaOwnerName2, schema2, schemaOwnerName1));
+        assertQuery(
+                "SELECT * FROM system.metadata.schemas_authorization WHERE catalog = 'memory'",
+                "VALUES('memory', '%s', 'USER', '%s')".formatted(schema1, schemaOwnerName2));
+        assertQuery(
+                "SELECT * FROM system.metadata.schemas_authorization WHERE catalog = 'memory_test'",
+                "VALUES('memory_test', '%s', 'USER', '%s')".formatted(schema2, schemaOwnerName1));
+
+        getQueryRunner().execute(schemaOwner2, "DROP SCHEMA memory.%s".formatted(schema1));
+        getQueryRunner().execute(schemaOwner1, "DROP SCHEMA memory_test.%s".formatted(schema2));
+        assertQueryReturnsEmptyResult("SELECT * FROM system.metadata.schemas_authorization");
+    }
+
+    @Test
+    public void testTablesAuthorization()
```

Comment:
add test for VIEW and MATERIALIZED VIEW

Reply:
VIEWS are covered by `testTablesAuthorization`  and MATERIALIZED VIEWS won't work without https://github.com/trinodb/trino/pull/25910/files#diff-438bae34efa7405d6c70ca5d7250d6348c6cab58176314eda60d24275d652c2f